dev-tools/docker/build_linux_jdk.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0 and the following additional limitation. Functionality enabled by the
+# files subject to the Elastic License 2.0 may only be used in production when
+# invoked by an Elasticsearch process with a license key installed that permits
+# use of machine learning features. You may not use this file except in
+# compliance with the Elastic License 2.0 and the foregoing additional
+# limitation.
+#
+
+# Builds the Docker image that can be used to run a subset of the ES
+# integration tests.
+#
+# This script is not intended to be run regularly.
+
+if [ `uname -m` != x86_64 ] ; then
+    echo "Native build images must be built on the correct hardware architecture"
+    echo "Required: x86_64, Current:" `uname -m`
```

Comment:
We'll need a similar image for `aarch64` soon too.

It might be worth thinking about the container naming here. With our existing images `aarch64` came along a long time after we first created the images, hence we have `ml_linux_build` and `ml_linux_native_aarch64_build`. But if we'd thought from the start that we'd be supporting multiple architectures we wouldn't have named them like this.

I think there are 2 options:

1. Fit with our current image naming, so `ml_cpp_jdk` -> `ml_cpp_linux_jdk`, and later introduce `ml_cpp_linux_aarch64_jdk`.
2. Use architecture in all names from the start, so `ml_cpp_jdk` -> `ml_cpp_linux_x86_64_jdk`, and later introduce `ml_cpp_linux_aarch64_jdk`.

Reply:
> 2\. Use architecture in all names from the start, so `ml_cpp_jdk` -> `ml_cpp_linux_x86_64_jdk`, and later introduce `ml_cpp_linux_aarch64_jdk`.

I like this latter naming scheme, even though it is at odds with the other existing images